tests/em/tdem/test_TDEM_sources.py

```diff
@@ -0,0 +1,204 @@
+        expected = np.array(
+            [0.0, 0.455054, 0.731059, 0.898464, 1.0, 0.75, 0.5, 0.25, 0.0, 0.0, 0.0]
+        )
+        assert_array_almost_equal(result, expected)
+
+
+class TestTrapezoidWaveform(unittest.TestCase):
+    @classmethod
+    def setUpClass(cls):
+        cls.times = np.linspace(start=0, stop=1e-2, num=11)
+
+    def test_waveform_with_symmetric_on_off(self):
+        trapezoid = TrapezoidWaveform(
+            ramp_on=np.r_[0.0, 2e-3], ramp_off=np.r_[6e-3, 8e-3]
+        )
+        result = [trapezoid.eval(t) for t in self.times]
+        expected = np.array([0.0, 0.5, 1.0, 1.0, 1.0, 1.0, 1.0, 0.5, 0.0, 0.0, 0.0])
+        assert_array_almost_equal(result, expected)
+
+    def test_waveform_with_asymmetric_on_off(self):
+        trapezoid = TrapezoidWaveform(
+            ramp_on=np.r_[0.0, 2e-3], ramp_off=np.r_[6e-3, 10e-3]
+        )
+        result = [trapezoid.eval(t) for t in self.times]
+        expected = np.array([0.0, 0.5, 1.0, 1.0, 1.0, 1.0, 1.0, 0.75, 0.5, 0.25, 0.0])
+        assert_array_almost_equal(result, expected)
+
+
+class TestTriangularWaveform(unittest.TestCase):
+    @classmethod
+    def setUpClass(cls):
+        cls.times = np.linspace(start=0, stop=1e-2, num=11)
+
+    def test_waveform_with_symmetric_on_off(self):
+        triangular = TriangularWaveform(peakTime=4e-3, offTime=8e-3)
+        result = [triangular.eval(t) for t in self.times]
+        expected = np.array([0.0, 0.25, 0.5, 0.75, 1.0, 0.75, 0.5, 0.25, 0.0, 0.0, 0.0])
+        assert_array_almost_equal(result, expected)
+
+    def test_waveform_with_asymmetric_on_off(self):
+        triangular = TriangularWaveform(peakTime=2e-3, offTime=6e-3)
+        result = [triangular.eval(t) for t in self.times]
+        expected = np.array([0.0, 0.5, 1.0, 0.75, 0.5, 0.25, 0.0, 0.0, 0.0, 0.0, 0.0])
+        assert_array_almost_equal(result, expected)
+
+
+class TestQuarterSineRampOnWaveform(unittest.TestCase):
+    @classmethod
+    def setUpClass(cls):
+        cls.times = np.linspace(start=0, stop=1e-2, num=11)
+
+    def test_waveform_with_plateau(self):
+        quarter_sine = QuarterSineRampOnWaveform(
+            ramp_on=np.r_[0.0, 3e-3], ramp_off=np.r_[7e-3, 1e-2]
+        )
+        result = [quarter_sine.eval(t) for t in self.times]
+        expected = np.array(
+            [0.0, 0.5, 0.866025, 1.0, 1.0, 1.0, 1.0, 1.0, 0.666667, 0.333333, 0.0]
+        )
+
+        assert_array_almost_equal(result, expected)
+
+    def test_waveform_without_plateau(self):
+        quarter_sine = QuarterSineRampOnWaveform(
+            ramp_on=np.r_[0.0, 5e-3], ramp_off=np.r_[5e-3, 1e-2]
+        )
+        result = [quarter_sine.eval(t) for t in self.times]
+        expected = np.array(
+            [0.0, 0.309017, 0.587785, 0.809017, 0.951057, 1.0, 0.8, 0.6, 0.4, 0.2, 0.0]
+        )
+
+        assert_array_almost_equal(result, expected)
+
+    def test_waveform_negative_plateau(self):
+        """TODO: should this throw a ValueError instead?"""
```

Comment:
Did you mean to leave this in?

Reply:
No... that was a reminder for myself to open a separate issue for simpeg. Thanks for catching it!